### ISSUE TITLE
WANTED! Installfest and Example Testers

### ISSUE THREAD

**Issue by broesamle** [rust-community/rustbridge#45]
Code examples that will be used in workshops (for beginners) should have been tested!

Your experiences testing the instructuins/examples will help establish a stable learning environment in the workshops:

- [ ] [setup your machine](https://github.com/rust-community/rustbridge/blob/master/workshops/installfest/user-guide.md)
- [ ] [beginners](https://github.com/rust-community/rustbridge/tree/master/workshops/beginners)
- [ ] url shortener #14
- [ ] [mondrian pattern](https://github.com/rust-community/rustbridge/tree/master/workshops/mondrian-pattern/codebase/mondpaint) #15 

Before you start, please get in contact via a post/issue so that the developers can coordinate effort and avoid testing of premature work!

It does not matter, which system you are workin on:
+ Windows 
+ mac OS
+ linux
  - debian-based
  - Ubuntu (sometimes differs from other Linuxes regarding developer packages)
. . .

**Comment by booyaa** [comment]
I'll ask my meetup group to test [setup your machine](https://github.com/rust-community/rustbridge/blob/master/workshops/installfest/user-guide.md) and url shortener #14. The latter we plan on iterating.

**Comment by aepsil0n** [comment]
So the beginners stuff on Windows requires `gcc.exe`. That is a somewhat tricky requirement for beginners.

**Comment by aepsil0n** [comment]
And the URL shortener setup does not work as is on any platform. There is a lifetime bug that needs fixing.

**Comment by broesamle** [comment]
Yes, I am aware of this .. on the long run there will (hopefully) be a more dependency-lightweight rust graphics environment available.

In the time availabke I could not work around this. 

I will try to write up a quick setup guide for MSYS+gcc (which woked well for me) as soon I am done with the workshop materials.
For the workshop on sunday we may have to support some windows people in person.

**Comment by broesamle** [comment]
thanks for testing, in any case!

**Comment by aepsil0n** [comment]
Yes, I have the impression that supporting in person is our only option at this point. And it's going to be fine ;)

**Comment by broesamle** [comment]
I thought about setting up a VM so that win users could just install virtual box and copy the image. 

avoids 64/32 bit pitfalls on windows which cost me some time abd are tricky to figure out under time pressure

any suggestions for a lightweight linux with a window maker that does not need gpu access are welcome...

-------- Ursprüngliche Nachricht --------
Von: Eduard Bopp <notifications@github.com> 
Datum: 16.09.2016  22:55  (GMT+01:00) 
An: rust-community/rustbridge <rustbridge@noreply.github.com> 
Cc: "Martin Br:osamle" <info@m-broe.de>,Author <author@noreply.github.com> 
Betreff: Re: [rust-community/rustbridge] WANTED! Installfest and Example Testers (#45) 
 
Yes, I have the impression that supporting in person is our only option at this point. And it's going to be fine ;)

**Comment by bruceadams** [comment]
A potential problem with a VM approach is that a VM needs a pretty hefty machine to run well. Our students might have old, slow machines.

**Comment by LeonardA-L** [comment]
I'm sorry if this looks like a shameless plug, but I do work at a company that provides just the solution to this. We have a beta for an online knowledge sharing platform, where people can create hands-on courses and tutorials. Everything runs on our back-end so learners can try their hands on technologies from an online IDE. It's based on docker and can run virtually any technology.

[Here's an example that runs rust](https://www.codingame.com/profile/1f902a3213291dfd0de802207d362bdf979991/course/sumprimesrust/lesson/sum-prime-numbers) and you can run basically any project with any dependency since contributors can package everything for the course on a [git repo](https://github.com/LeonardA-L/rust-codingame-course).

It's free for everyone, teachers and learners, it might be a good solution to host hands-on workshops without having to worry about installations, what do you think?

**Comment by loganmzz** [comment]
Maybe building a bootable image that can be used as a bootable device (USB key) or as VM can help. I think having explanation for full installation from scratch is still mandatory.

And gcc requirements is not true as you can install the MS-based toolchain. It only requires administrative rights such as VM supervisor installation. This last one is much hard to handle that just installing MSYS with gcc or MS SDK.

Moreover for Windows installation instruction, rustup isn't mention but works perfectly !

Note 1: links into first issue message are broken after b5ad4fc2b4602b910b9ac6ffae3ba92a5f315e1f

Note 2: Atom is the only mentionned tool but european have lot of problem with it due to some keyboard management issue. There was supposed to be fixed but some users still encounter some problems. From my personal experience, Atom is (strangely) much more slower than VS code.
